File: src/ExposureProtectedMember.java
public class ExposureProtectedMember {
    public static void main(String[] args) {
        SubClass subClass = new SubClass();
        subClass.print();
    }
}
class SuperClass {
    protected int m = 1;
}

class SubClass extends SuperClass {
    public int m = 2;

    public void print() {
        System.out.println(super.m);
        System.out.println(this.m);
    }
}

File: src/ExposureProtectedMember.ts
class SuperClass {
    protected m = 1;
}
class SubClass extends SuperClass {
    // No modifier, so default is 'public'
    public print = () => {
        console.log(super.m);
        console.log(this.m);
    }
    // public m = 2;
}
const subClass = new SubClass();
subClass.print();